lib/flutter_periculum.dart

```diff
@@ -267,4 +274,218 @@ class FlutterPericulum {
+          'Authorization': 'Bearer $token',
+        },
+      );
+
+      var result = response.body;
+      map = json.decode(result);
+      MobileAnalysisResponseV2 getMobileV2InsightOverview =
+          MobileAnalysisResponseV2.fromJson(map);
+      return getMobileV2InsightOverview;
+    } on FormatException catch (_) {
+      log("FormatException: Invalid Acccess Token or Insights Key");
+      rethrow;
+    } on HttpException catch (_) {
+      log("HttpException: Check your connection");
+      rethrow;
+    } catch (e) {
+      log(e.toString());
+      rethrow;
+    }
+  }
+
+  static Future<BankResponse> getMobileV2BanksForInsightOverviewKey({
+    required String token,
+    required String insightsOverviewKey,
+  }) async {
+    final uri = Uri.parse('$BASE_URL/mobile/v2/$insightsOverviewKey/banks');
+
+    var client = http.Client();
+    Map<String, dynamic> map;
+    var response;
+    try {
+      response = await client.get(
+        uri,
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': 'Bearer $token',
+        },
+      );
+
+      var result = response.body;
+      map = json.decode(result);
+      BankResponse bankResponse = BankResponse.fromJson(map);
+      return bankResponse;
+    } on FormatException catch (_) {
+      log("FormatException: Invalid Acccess Token or Insights Key");
+      rethrow;
+    } on HttpException catch (_) {
+      log("HttpException: Check your connection");
+      rethrow;
+    } catch (e) {
+      log(e.toString());
+      rethrow;
+    }
+  }
+
+  static Future<LendersResponse> getMobileV2LendersForInsightOverviewKey({
+    required String token,
+    required String insightsOverviewKey,
+  }) async {
+    final uri = Uri.parse('$BASE_URL/mobile/v2/$insightsOverviewKey/lenders');
+
+    var client = http.Client();
+    Map<String, dynamic> map;
+    var response;
+    try {
+      response = await client.get(
+        uri,
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': 'Bearer $token',
+        },
+      );
+
+      var result = response.body;
+      map = json.decode(result);
+      LendersResponse lendersResponse = LendersResponse.fromJson(map);
+      return lendersResponse;
+    } on FormatException catch (_) {
+      log("FormatException: Invalid Acccess Token or Insights Key");
+      rethrow;
+    } on HttpException catch (_) {
+      log("HttpException: Check your connection");
+      rethrow;
+    } catch (e) {
+      log(e.toString());
+      rethrow;
+    }
+  }
+
+  static Future<List<TransactionResponse>>
+      getMobileV2TransactionsByBankAddressAndAccountNumberForInsightOverviewKey({
+    required String token,
+    required String insightsOverviewKey,
+    required String smsAddress,
+    required String accountKey,
+  }) async {
+    final uri = Uri.parse(
+        '$BASE_URL/mobile/v2/$insightsOverviewKey/banks/$smsAddress/$accountKey/transactions');
```

Comment:
this should be bankKey instead of smsAddress

Reply:
Oh okay